projects/js-packages/partner-coupon/hooks.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect } from 'react';
+import { getRedirectUrl } from '@automattic/jetpack-components';
+
+/**
+ * Handle partner coupon redeem analytics and clicks.
+ *
+ * @param {object} partnerCoupon - Partner coupon details.
+ * @param {string} siteRawUrl - Site's raw URL.
+ * @param {object} connectionStatus - Connection status.
+ * @param {object} tracksUserData - Tracks user data.
+ * @param {object} analytics - Analytics.
+ * @returns {Function} Click handler for coupon redemption.
+ */
+export function usePartnerCouponRedemption(
+	partnerCoupon,
+	siteRawUrl,
+	connectionStatus,
+	tracksUserData,
+	analytics
+) {
+	useEffect( () => {
+		if ( tracksUserData && 'object' === typeof analytics ) {
+			analytics.tracks.recordEvent( 'jetpack_partner_coupon_redeem_view', {
+				coupon: partnerCoupon.coupon_code,
+				partner: partnerCoupon.partner.prefix,
+				preset: partnerCoupon.preset,
+				// This is expected to always be "yes" since we do not track users
+				// before they have connected and agreed to our ToS, but we'll leave
+				// it in for historical reasons if this change some day.
+				connected: connectionStatus.isRegistered ? 'yes' : 'no',
+			} );
+		}
+	}, [ analytics, tracksUserData, connectionStatus, partnerCoupon ] );
```

Comment:
Oi! I just realised that this tracking code is only triggered on click events, so we don't actually get the "view vs click" data to compare how effective it is.

Reply:
> this tracking code is only triggered on click events

How so? Is this something about how `analytics` works or ...?

Reply:
The `usePartnerCouponRedemption` custom hook returns a click handler callback but the above code uses the `useEffect` hook which will execute and is not bound to the click handler. You can test this by adding a console log inside the `useEffect` callback and it will log whenever the components that use the hook render (or when any of the hook dependencies change of course).

Reply:
Ah! Thank you for the explanation Atanas, makes perfect sense 😄